### PR TITLE
Fix/comment component refactoring

### PR DESCRIPTION
**Pull request by DmitriiMelnikOnix**
https://github.com/flanksource/flanksource-ui/issues/171

### CODE REVIEW

File: src/components/HypothesisBuilder/comments.js [hunk 1]
```diff
@@ -69,69 +32,39 @@ export function CommentsSection({
...
-      },
-      [otherText[0]]
-    );
-  };
+  const commentData = useMemo(
+    () => [
+      {
+        data: users,
+        markup: "@[__display__](user:__id__)",
+        trigger: "@"
```
Comment (moshloop) [inline src/components/HypothesisBuilder/comments.js:53]: Can we move markup and trigger as attributes on Comment Input and then default them ?

Comment (DmitriiMelnikOnix) [reply src/components/HypothesisBuilder/comments.js:53]: We can if we use only mentions of users. But if you need to make tags mentions and user mentions then the current example will be better

Comment (moshloop) [reply src/components/HypothesisBuilder/comments.js:53]: Even if we add support for tags, it would be in addition `#` to users not instead of

Comment (DmitriiMelnikOnix) [reply src/components/HypothesisBuilder/comments.js:53]: Do you suggest doing the following?
```
<CommentInput data={data} markup="@[__display__](user:__id__)" trigger="@" ... />
```
But how then to make support a few triggers and markup?
In this case, we can use either @ or # trigger. But not both at the same time

Comment (moshloop) [reply src/components/HypothesisBuilder/comments.js:53]: I suggest that interface but default the markup and trigger, at a later point in the future if we decide to handle tags we can come up with a solution then

Comment (DmitriiMelnikOnix) [reply src/components/HypothesisBuilder/comments.js]: Ok I updated interface

**Pull request merged**